handler-egsp/src/main/java/com/expediagroup/streamplatform/streamregistry/handler/EgspKafkaStreamHandler.java

```diff
@@ -15,31 +15,71 @@
  */
 package com.expediagroup.streamplatform.streamregistry.handler;
 
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
 import org.springframework.stereotype.Component;
 
 import com.expediagroup.streamplatform.streamregistry.model.Specification;
 import com.expediagroup.streamplatform.streamregistry.model.Stream;
 
 @Component
-public class EgspKafkaStreamHandler<E extends Stream> implements Handler<E> {
-
+public class EgspKafkaStreamHandler implements Handler<Stream> {
   @Override
   public String type() {
     return "egsp.kafka";
   }
 
   @Override
-  public Class target() {
+  public Class<Stream> target() {
     return Stream.class;
   }
 
   @Override
-  public Specification handleInsert(E stream) throws HandlerException {
-    return stream.getSpecification();
+  public Specification handleInsert(Stream stream) {
+    return handle(stream);
   }
 
   @Override
-  public Specification handleUpdate(E stream, E existing) throws HandlerException {
-    return stream.getSpecification();
+  public Specification handleUpdate(Stream stream, Stream existing) {
+    return handle(stream);
+  }
+
+  private Specification handle(Stream stream) {
+    Specification specification = stream.getSpecification();
+    ObjectNode configuration = specification.getConfiguration();
+
+    ObjectNode log = (ObjectNode) configuration.get("log");
+    checkNotNull(log, "log must not be null");
+
+    validateGreaterThanZero(log, "partitions");
+    validateGreaterThanZero(log, "replicationFactor");
```

Comment:
Can we change this to `replication.factor` to match all the other config naming conventions?
```suggestion
    validateGreaterThanZero(log, "replication.factor");
```

Reply:
done